### PR TITLE
ci: add release workflow

### PR DESCRIPTION
**Pull request by xrmx**
@elastic/observablt-robots This needs settings some secrets right?

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
```
Comment (xrmx) [inline .github/workflows/release.yml:6]: Ah tags here does not start with v

Comment (xrmx) [inline .github/workflows/release.yml:6]:
```suggestion
      - "*.*.*"
```

Comment (reakaleek) [reply .github/workflows/release.yml:6]: What do you think about something like `[0-9]+.[0-9]+.[0-9]+`?

**Comment by reakaleek** [comment]
> This needs settings some secrets right?

@xrmx If I understand correctly, the `pypa/gh-action-pypi-publish` action uses OIDC to authenticate. Hence, if everything is set up correctly on the pypi side. I don't see a need for any secrets.

Unless I'm missing something?

**Review by reakaleek — DISMISSED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (xrmx) [reply .github/workflows/release.yml:6]: Yeah, that's better

**Review by reakaleek — COMMENTED** [review]
LGTM.

An enhancement we could add in a follow-up is moving the packages job into a reusable workflow and then we can run it both here in the release and in another workflow with a pull_request or push to main trigger to verify it's working before the release.

**Comment by xrmx** [comment]
> > This needs settings some secrets right?
> 
> @xrmx If I understand correctly, the `pypa/gh-action-pypi-publish` action uses OIDC to authenticate. Hence, if everything is set up correctly on the pypi side. I don't see a need for any secrets.
> 
> Unless I'm missing something?

The elastic-apm-team account is listed as mantainer for this package

**Comment by reakaleek** [comment]
> > > This needs settings some secrets right?
> > 
> > 
> > @xrmx If I understand correctly, the `pypa/gh-action-pypi-publish` action uses OIDC to authenticate. Hence, if everything is set up correctly on the pypi side. I don't see a need for any secrets.
> > Unless I'm missing something?
> 
> The elastic-apm-team account is listed as mantainer for this package

Sorry, I don't know what that means. I haven't worked with pypi yet. But I assume you must set up a trust relationship between the pypi package and this repository.

**Review by v1v — APPROVED** [review]
LGTM.

**Comment by v1v** [comment]
> Sorry, I don't know what that means. I haven't worked with pypi yet. But I assume you must set up a trust relationship between the pypi package and this repository.

I'll double check this and document what I've done, so we don't forget it :)

**Review by v1v — APPROVED** [review]
Do you want to support test-pypi too? similarly done in https://github.com/elastic/apm-agent-python/blob/dea50dd92a2cef299cbff08e339b4459a72ad453/.github/workflows/release.yml#L53-L57

**Comment by xrmx** [comment]
> Do you want to support test-pypi too? similarly done in https://github.com/elastic/apm-agent-python/blob/dea50dd92a2cef299cbff08e339b4459a72ad453/.github/workflows/release.yml#L53-L57

Nah, the CI here is much simpler :)

**Review by reakaleek — APPROVED** [review]

**Pull request merged**